File: app/src/main/java/bb/incognito/view/AddCocktailFragment.java
package bb.incognito.view;
import android.app.DialogFragment;
import android.content.Context;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.WindowManager;
import android.view.inputmethod.InputMethodManager;

import bb.incognito.R;
import bb.incognito.databinding.FragmentAddCocktailBinding;
import bb.incognito.model.Guest;
import bb.incognito.repositories.CocktailRepository;
import bb.incognito.view.adapter.CocktailAdapter;
import bb.incognito.viewModel.AddCocktailVM;

import com.google.android.gms.plus.PlusOneButton;

public class AddCocktailFragment extends DialogFragment {

    FragmentAddCocktailBinding fragmentAddCocktailBinding;
    static CocktailRepository cocktailRepository;
    public AddCocktailFragment(){}
    static Guest guest;

    public static AddCocktailFragment newInstance(boolean edit, int pos, CocktailRepository cR, Guest _guest){
        AddCocktailFragment addCocktailFragment = new AddCocktailFragment();
        Bundle args = new Bundle();
        args.putBoolean("edit", edit);
        args.putInt("pos", pos);
        addCocktailFragment.setArguments(args);
        cocktailRepository = cR;
        guest = _guest;
        return addCocktailFragment;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        fragmentAddCocktailBinding = DataBindingUtil.inflate(inflater,
                R.layout.fragment_add_cocktail, container, false);

        int pos = getArguments().getInt("pos");
        boolean edit = getArguments().getBoolean("edit");

        fragmentAddCocktailBinding.setViewModel(new AddCocktailVM(this,edit, pos, cocktailRepository, guest));
        return fragmentAddCocktailBinding.getRoot();
    }

    @Override
    public void onResume(){
        super.onResume();

        fragmentAddCocktailBinding.newName.requestFocus();
        getDialog().getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE);
    }

    @Override
    public void dismiss(){
        ((InputMethodManager) getDialog().getContext().getSystemService(Context.INPUT_METHOD_SERVICE))
                .hideSoftInputFromWindow(getView().getWindowToken(), 0);
        super.dismiss();
    }
}

File: app/src/main/java/bb/incognito/view/AddCocktailFragment.kt
package bb.incognito.view

import android.arch.lifecycle.Observer
import android.support.v4.app.Fragment
import android.arch.lifecycle.ViewModelProviders
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.SearchView
import bb.incognito.R
import bb.incognito.databinding.FragmentAddCocktailBinding
import bb.incognito.model.Cocktail
import bb.incognito.model.Guest
import bb.incognito.model.GuestWithCocktails
import bb.incognito.repositories.CocktailRepository
import bb.incognito.repositories.GuestRepository
import bb.incognito.repositories.GuestWithCocktailsRepository
import bb.incognito.view.adapter.CocktailAdapter
import bb.incognito.viewModel.CocktailsViewModel

class AddCocktailFragment : Fragment(), SearchView.OnQueryTextListener {

    internal var fragmentAddCocktailBinding: FragmentAddCocktailBinding? = null
    var cocktailAdapter: CocktailAdapter? = null
    var cocktailRepository : CocktailRepository? = null
    private var sv: SearchView? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        fragmentAddCocktailBinding = DataBindingUtil.inflate(inflater,
                R.layout.fragment_add_cocktail, container, false)
        cocktailRepository = CocktailRepository(activity.application)

        var viewModel = ViewModelProviders.of(this).get(CocktailsViewModel::class.java)
        cocktailAdapter = CocktailAdapter() //get data of cocktails. from guest
        cocktailAdapter?.setCheckable(true)
        fragmentAddCocktailBinding!!.list.adapter = cocktailAdapter
        fragmentAddCocktailBinding!!.list.layoutManager = LinearLayoutManager(context)

        viewModel.allCocktails.observe(this,
                Observer<List<Cocktail>> { cocktails -> cocktailAdapter!!.cocktailList = cocktails })
        sv = activity.findViewById(R.id.search)

        fragmentAddCocktailBinding?.cancelButton?.setOnClickListener {
            fragmentManager.popBackStack()
        }

        fragmentAddCocktailBinding?.acceptButton?.setOnClickListener {
            var guestWithCocktailRepository = GuestWithCocktailsRepository(activity.application)
            var guestRepository = GuestRepository(activity.application)
            var checkedCocktails = cocktailAdapter!!.checkedCocktails

            //guestRepository.insert(guest)
            //guestWithCocktailRepository.insert(guest)
            fragmentManager.popBackStack()
        }

        sv!!.setOnQueryTextListener(this)

        return fragmentAddCocktailBinding!!.root
    }

    override fun onQueryTextSubmit(s: String): Boolean {
        cocktailAdapter!!.filter.filter(s)
        return false
    }

    override fun onQueryTextChange(s: String): Boolean {
        cocktailAdapter!!.filter.filter(s)
        return false
    }

    companion object {
        var guest : GuestWithCocktails? = null
        fun launch(g: GuestWithCocktails) : Fragment{
            guest = g
            return AddCocktailFragment()
        }
    }
}
